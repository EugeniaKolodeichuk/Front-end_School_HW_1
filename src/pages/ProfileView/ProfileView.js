import { useParams } from 'react-router';
import React, { useState, useEffect } from 'react';
import styles from './ProfileView.module.css';
import Loader from '../../components/Loader/Loader';
import { getUserFeed, getUserInfo } from '../../service/app';
import UserVideos from '../../components/UserVideos/UserVideos';
import UserInfo from '../../components/UserInfo/UserInfo';

export default function ProfileView() {
  const { uniqueId } = useParams();

  const [userFeed, setUserFeed] = useState([]);
  const [userInfo, setUserInfo] = useState([]);

  useEffect(() => {
    try {
      getUserInfo(uniqueId).then(data => {
        setUserInfo(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserInfo]);

  useEffect(() => {
    try {
      getUserFeed(uniqueId).then(data => {
        setUserFeed(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserFeed]);

  return (
    <div key={userInfo.id} data-testid="profile">
      <UserInfo
        uniqueId={userInfo.uniqueId}
        nickname={userInfo.nickname}
        avatarMedium={userInfo.avatarMedium}
        signature={userInfo.signature}
      />
      <div className={styles.wrap} key={userFeed.id}>
        {userFeed > 0 ? (
          userFeed.map(user => <UserVideos key={user.createTime} user={user} />)
        ) : (
          <Loader />
        )}
      </div>
    </div>
  );
}
