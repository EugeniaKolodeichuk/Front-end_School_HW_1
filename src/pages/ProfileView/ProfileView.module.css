.wrap {
    max-width: 1300px;
    margin: 0 auto;
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
}