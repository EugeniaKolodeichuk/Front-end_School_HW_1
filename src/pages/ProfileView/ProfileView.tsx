import { useParams } from 'react-router';
import { useState, useEffect } from 'react';
import StyledWrap from './StyledProfileView';
import Loader from '../../components/Loader/Loader';
import { getUserFeed, getUserInfo } from '../../service/app';
import UserVideos from '../../components/UserVideos/UserVideos';
import UserInfo from '../../components/UserInfo/UserInfo';
import { IUserInfo } from '../../types/userTypes';

export default function ProfileView() {
  const { uniqueId } = useParams<{ uniqueId: string }>();

  const [userFeed, setUserFeed] = useState<IUserInfo[]>([]);
  const [userInfo, setUserInfo] = useState<IUserInfo | null>(null);

  useEffect(() => {
    try {
      getUserInfo(uniqueId).then(data => {
        setUserInfo(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserInfo]);

  useEffect(() => {
    try {
      getUserFeed(uniqueId).then(data => {
        setUserFeed(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserFeed]);
  if (!userInfo) {
    return <Loader />;
  }
  return (
    <div key={userInfo.id} data-testid="profile">
      <UserInfo
        uniqueId={userInfo.uniqueId}
        nickname={userInfo.nickname}
        avatarMedium={userInfo.avatarMedium}
        signature={userInfo.signature}
      />
      <StyledWrap key={userInfo.id}>
        {userFeed ? (
          userFeed.map((user: IUserInfo) => (
            <UserVideos key={user.createTime} user={user} />
          ))
        ) : (
          <Loader />
        )}
      </StyledWrap>
    </div>
  );
}
