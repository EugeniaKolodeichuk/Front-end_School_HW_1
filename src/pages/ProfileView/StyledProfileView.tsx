import styled from 'styled-components';

const StyledWrap = styled.div`
  max-width: 1300px;
  margin: 0 auto;
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
`;

export default StyledWrap;
