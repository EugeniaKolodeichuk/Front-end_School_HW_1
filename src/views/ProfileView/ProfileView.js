import { useParams } from 'react-router';
import React, { useState, useEffect } from 'react';
import defaultImage from '../../images/default.jpg';
import styles from './ProfileView.module.css';
import Loader from '../../components/Loader/Loader';
import { getUserFeed, getUserInfo } from '../../service/app';

export default function ProfileView() {
  const { uniqueId } = useParams();

  const [userFeed, setUserFeed] = useState([]);
  const [userInfo, setUserInfo] = useState([]);

  useEffect(() => {
    try {
      getUserInfo(uniqueId).then(data => {
        setUserInfo(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserInfo]);

  useEffect(() => {
    try {
      getUserFeed(uniqueId).then(data => {
        setUserFeed(data);
      });
    } catch (error) {
      console.error(error);
    }
  }, [setUserFeed]);

  return (
    <div key={userInfo.id}>
      {userInfo && (
        <div key={userInfo.uniqueId} className={styles.profile}>
          <div className={styles.list_noorder}>
            {userInfo.avatarThumb ? (
              <img
                key={userInfo.nickname}
                width="150px"
                src={userInfo.avatarMedium}
                alt={userInfo.nickname}
                className={styles.avatar}
              />
            ) : (
              <img width="150px" src={defaultImage} alt={userInfo.nickname} />
            )}
            <h2 className={styles.title}>{userInfo.nickname}</h2>
            <p>{userInfo.signature}</p>
          </div>
          <h2 className={styles.title}>
            Other posts created by {userInfo.nickname}
          </h2>
        </div>
      )}
      <div className={styles.wrap} key={userFeed.id}>
        {userFeed.length ? (
          userFeed.map(user => (
            <div key={user.createTime} className={styles.list_noorder}>
              <p>
                <b>{user.stats.playCount}</b> views
              </p>
              {user.video.originCover ? (
                <img
                  key={user.video.id}
                  src={user.video.originCover}
                  alt={user.desc}
                  className={styles.main_element}
                />
              ) : (
                <img width="150px" src={defaultImage} alt={userInfo.nickname} />
              )}

              <p>{user.desc}</p>
            </div>
          ))
        ) : (
          <Loader />
        )}
      </div>
    </div>
  );
}
