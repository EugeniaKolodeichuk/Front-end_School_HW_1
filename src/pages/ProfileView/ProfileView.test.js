import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ProfileView from './ProfileView';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('#ProfileView.js', () => {
  it('render profile page', () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <ProfileView />
      </Router>,
      container,
    );
    expect(container).toMatchSnapshot();
  });

  it('check for special text at the page', () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <ProfileView />
      </Router>,
    );

    expect(screen.getByText(/other posts created by/i)).toBeInTheDocument();
  });
});
