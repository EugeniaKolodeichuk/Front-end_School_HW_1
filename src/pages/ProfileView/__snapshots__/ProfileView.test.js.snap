// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProfileView.js render profile page 1`] = `<div />`;
